lib/color_mask.py

```diff
@@ -1,163 +1,225 @@
+    numpy arrays. 
+    """
+    lower = np.array(d["lower"])
+    upper = np.array(d["upper"])
+    return lower, upper
+
+
+def color_histogram(chan):
+    """
+    After splitting an image into three channels (paths) of an HSV, call this
+    function with one of these paths to get a _histogram_ array, where the `x`
+    axis is one of 256 color values, and the `y` axis is the number of pixels
+    in the image that match that value.
+    """
+    hist = cv2.calcHist([chan], [0], None, [256], [0, 256])
+    color_frequency = [p[0] for p in hist]
+
+    # The numpy array (np.array) allows us to use these two demensional array
+    return np.array(color_frequency)
+
+
+def image_colors(image):
+    """
+    Returns a list of three histogram arrays, one for each of the HSV channels
+    in the image given.
+    """
+    hsv = cv2.cvtColor(image, cv2.COLOR_BGR2HSV)
+    # Split into the three channels of hue, saturation, and value.
+    # We then initialize a tuple of strings representing the colors.
+    chans = cv2.split(hsv)
+
+    return [color_histogram(chan) for chan in chans]
+
+
+def color_range(image):
+    """
+    Smooth values (from graph.py) and with max_peak_deviation finds the most
+    frequent colors as a list of two values that form a _range_ of the most
+    prominent color in image.
+
+    It does this by making a bell curve from the images' most frequently used
+    color, as well as its neighbors, by sloping downwards on both sides to the
+    curve's local mins.
+    """
+    min_color = []
+    max_color = []
+
+    for hist_channel in image_colors(image):
+        l, u = top_bell(smooth(hist_channel))
+        # The standard deviation creates too small of a range, about 50 is
+        # needed to be added and subtracted from the min and max values to
+        # account for changes in light
+        min_color.append(l)
+        max_color.append(u)
+
+    lower = np.asarray(min_color)  # Able to use two demensional array
+    upper = np.asarray(max_color)
+
+    return(lower, upper)  # returns lowest and highest most frequent values
+
+
+def get_mask(hsv_img, lower, upper):
+    """
+    Given an image and color range, return a simplified masked image.
+    """
+    thresh = cv2.inRange(hsv_img, lower, upper)
+
+    # perform some clean up before contour operations
+    # thresh = cv2.erode(thresh, None, iterations=2)
+    # thresh = cv2.dilate(thresh, None, iterations=2)
+
+    return thresh
+
+
+def get_contours(img):
+    """
+    Contours are a curve joining all the continuous points along the
+    boundary of the same color or intensity. The list of contours are
+    sorted based on size from smallest to largest.
+    """
+    # _ = image that is ignored
+    _, contours, hierarchy = cv2.findContours(img.copy(), cv2.RETR_EXTERNAL,
+                                              cv2.CHAIN_APPROX_SIMPLE)
+    return sorted(contours, key=cv2.contourArea)
+
+
+def single_target(img, orig=[]):

```

Comment:
I think it actually makes more sense to move this `single_target` function with the other targeting functions you have in `target_tracker.py`